src/compiler/typescript_to_rust/generate_rust_canister.ts

```diff
@@ -24,61 +28,64 @@ export function generateRustCanister(
 
     const compilerInfoPath = join(
         canisterPath,
-        rootPath,
+        canisterConfig.root,
         'azle_generate',
         'compiler_info.json'
     );
 
     writeFileSync(compilerInfoPath, JSON.stringify(compilerInfo));
 
-    const azleGenerateResult = runAzleGenerate(
-        'compiler_info.json',
-        canisterPath,
-        rootPath
-    );
-
-    if (!ok(azleGenerateResult)) {
-        return Err(azleGenerateResult.err);
-    }
-
-    const unformattedLibFile = azleGenerateResult.ok;
+    runAzleGenerate('compiler_info.json', canisterPath, canisterConfig);
 
-    writeFileSync(`${canisterPath}/${rootPath}/src/lib.rs`, unformattedLibFile);
-
-    const runRustFmtResult = runRustFmt(unformattedLibFile, canisterPath, {
-        rootPath
+    runRustFmt(canisterPath, {
+        rootPath: canisterConfig.root
     });
 
-    if (!ok(runRustFmtResult)) {
-        return Err(runRustFmtResult.err);
-    }
-
-    const formattedLibFile = runRustFmtResult.ok;
-
-    writeFileSync(`${canisterPath}/${rootPath}/src/lib.rs`, formattedLibFile);
-
-    if (isVerboseMode()) {
-        console.info(
-            `Wrote formatted lib.rs file to ${canisterPath}/${rootPath}/src/lib.rs\n`
-        );
-    }
     return Ok(undefined);
 }
 
 function runAzleGenerate(
     compilerInfoPath: string,
     canisterPath: string,
-    rootPath: string
-): Result<Rust, AzleError> {
-    if (isVerboseMode()) {
-        console.info('running azle_generate');
+    canisterConfig: JSCanisterConfig
+) {
+    // TODO to cut the time in half here, if we can just generate the binary
+    // TODO and then after that use the binary, that could work
+    // TODO the problem is that this binary is stored in a global target dir
+    // TODO we might need to have an azle-version-specific dir for the azle_generate binary
+    // TODO maybe that's fine and easy though
+    const buildResult = spawnSync(`${GLOBAL_AZLE_BIN_DIR}/cargo`, ['build'], {
+        cwd: `${canisterPath}/${canisterConfig.root}/azle_generate`,
+        stdio: [
+            'inherit',
+            isVerboseMode() ? 'inherit' : 'pipe',
+            isVerboseMode() ? 'inherit' : 'pipe'
+        ],
+        env: {
+            ...process.env,
+            CARGO_TARGET_DIR: GLOBAL_AZLE_TARGET_DIR,
+            CARGO_HOME: GLOBAL_AZLE_RUST_DIR,
+            RUSTUP_HOME: GLOBAL_AZLE_RUST_DIR
+        }
+    });
+
+    // TODO should we handle presenting errors to the user somehow here?
+    // TODO maybe we should tell them to just rerun with verbose if the status code is not 0
+    if (buildResult.status !== null && buildResult.status !== 0) {
```

Comment:
Make sure to handle error and status code...handle any possible failure within reason